providers/base/bin/snap_confinement_test.py

```diff
@@ -17,64 +17,173 @@
+        """
+        data = Snapd().get_system_info()
 
-    confinement = data["confinement"]
-    if confinement == "strict":
-        print("Test PASS")
-        return 0
+        confinement = data["confinement"]
+        if confinement == "strict":
+            print("System confinement is \"strict\"")
+            print("Test PASS")
+            return 0
 
-    sandbox_features = data["sandbox-features"]
-    sandbox_features_output = (
-        "\nOUTPUT: confinement: {}\nOUTPUT: sandbox-features:\n{}".format(
-            confinement, json.dumps(sandbox_features, indent=2)
+        sandbox_features = data["sandbox-features"]
+        sandbox_features_output = (
+            "\nOUTPUT: confinement: {}\nOUTPUT: sandbox-features:\n{}".format(
+                confinement, json.dumps(sandbox_features, indent=2)
+            )
         )
-    )
-
-    missing_features = []
-    if "apparmor" not in sandbox_features:
-        logging.error("Cannot find 'apparmor' in sandbox-features")
-    else:
-        for feature in features_should_include:
-            if feature not in sandbox_features["apparmor"]:
-                missing_features.append(feature)
-
-    if missing_features:
-        logging.error("Cannot find '{}' in apparmor".format(missing_features))
-
-    categories_to_check = ["mount", "udev"]
-    for category in categories_to_check:
-        if category not in sandbox_features:
+
+        missing_features = []
+        if "apparmor" not in sandbox_features:
+            logging.error("Cannot find 'apparmor' in sandbox-features")
+        else:
+            for feature in self.features_should_include:
+                if feature not in sandbox_features["apparmor"]:
+                    missing_features.append(feature)
+
+        if missing_features:
             logging.error(
-                "Cannot find '{}' in sandbox-features".format(category)
+                "Cannot find '%s' in apparmor", missing_features
             )
-            break
-        for feature in sandbox_features[category]:
-            if "cgroup-v2" in feature:
-                logging.error("cgroup({}) must NOT be v2".format(feature))
 
-    return sandbox_features_output
+        categories_to_check = ["mount", "udev"]
+        for category in categories_to_check:
+            if category not in sandbox_features:
+                logging.error(
+                    "Cannot find '%s' in sandbox-features", category
+                )
+                break
+            for feature in sandbox_features[category]:
+                if "cgroup-v2" in feature:
+                    logging.error("cgroup(%s) must NOT be v2", feature)
+
+        return sandbox_features_output
+
+
+class SnapsConfinement:
+    """
+    Test the confinement status of all installed snaps.
+
+    Attributes:
+        whitelist_snaps (list): A list of snap names or regex patterns
+            that are exempted from the confinement check.
+    """
+
+    whitelist_snaps = [
```

Comment:
we're following modern naming schemes in which "allowlist" is preferred.

Reply:
Good suggestion! I'll modify the naming in the next commit.